src/js/controller/settings/exportimage/GifExportController.js

```diff
@@ -189,4 +192,67 @@
     }
     return url;
   };
+
+  ns.GifExportController.prototype.createGifExportDesc = function (i18n) {
+    var templateData = {
+      description: i18n.gifExportSettingSectionDesc(),
+    };
+    var templateId = 'gif-export-desc-template';
+    return pskl.utils.Template.fillInTemplate(templateId, templateData);
+  };
+
+  ns.GifExportController.prototype.createGifExportWarning = function (i18n) {
+    var templateData = {
+      description: i18n.gifExportSettingSectionWarningMessage(),
+    };
+    var templateId = 'gif-export-warning-template';
+    return pskl.utils.Template.fillInTemplate(templateId, templateData);
+  };
+
+  ns.GifExportController.prototype.createGifExportLoopRepeatedly = function (i18n) {
+    var templateData = {
+      label: i18n.gifExportSettingSectionLoopRepeatedly(),
+      tooltipTitle: i18n.gifExportSettingSectionUncheckToPlay()
+    };
+    var templateId = 'gif-export-loop-repeatedly-template';
+    return pskl.utils.Template.fillInTemplate(templateId, templateData);
+  };
+
+  ns.GifExportController.prototype.createGifExportDownloadSection = function (i18n) {
+    var templateData = {
+      codeDotOrgClass: '',
+      id: 'download-gif-section',
+      cssClass: 'gif-download-button',
+      buttonText: i18n.gifExportSettingSectionDownload(),
+      description: i18n.gifExportSettingSectionDownloadDesc()
+    };
+    var templateId = 'gif-export-upload-download-template';
+    return pskl.utils.Template.fillInTemplate(templateId, templateData);
+  };
+
+  ns.GifExportController.prototype.createGifExportUploadSection = function (i18n) {
+    var templateData = {
+      codeDotOrgClass: 'upload-gif-gamelab',
+      id: 'upload-gif-section',
+      cssClass: 'gif-upload-button',
+      buttonText: i18n.gifExportSettingSectionUpload(),
+      description: i18n.gifExportSettingSectionUploadDesc()
+    };
+    var templateId = 'gif-export-upload-download-template';
+    return pskl.utils.Template.fillInTemplate(templateId, templateData);
+  };
+
+  /**
+  * @private
+  */
+  ns.GifExportController.prototype.createGifExportDom_ = function (i18n) {
+    var html = '';
+    $('#export-info').html(this.createGifExportDesc(i18n));
+    $('#gif-export-warning').html(this.createGifExportWarning(i18n));
+    $('#loop-repeatedly-section').html(this.createGifExportLoopRepeatedly(i18n));
+    html += this.createGifExportDownloadSection(i18n);
+    html += this.createGifExportUploadSection(i18n);
+    html += '<div class=\'gif-upload\'><div class=\'gif-export-preview\'></div><div class=\'gif-upload-status\'></div></div>';
+    $('#gif-export-actions-section').html(html);
```

Comment:
minor nit, move `var html = '';` declaration closer to where it is used.
```suggestion
    $('#export-info').html(this.createGifExportDesc(i18n));
    $('#gif-export-warning').html(this.createGifExportWarning(i18n));
    $('#loop-repeatedly-section').html(this.createGifExportLoopRepeatedly(i18n));
    var html = '';
    html += this.createGifExportDownloadSection(i18n);
    html += this.createGifExportUploadSection(i18n);
    html += '<div class=\'gif-upload\'><div class=\'gif-export-preview\'></div><div class=\'gif-upload-status\'></div></div>';
    $('#gif-export-actions-section').html(html);
```